homeassistant/components/emoncms/config_flow.py

```diff
@@ -0,0 +1,209 @@
+)
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_ID,
+    CONF_SCAN_INTERVAL,
+    CONF_UNIT_OF_MEASUREMENT,
+    CONF_URL,
+    CONF_VALUE_TEMPLATE,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.selector import selector
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    CONF_EXCLUDE_FEEDID,
+    CONF_FEED_LIST,
+    CONF_ONLY_INCLUDE_FEEDID,
+    CONF_SENSOR_NAMES,
+    DOMAIN,
+    LOGGER,
+)
+
+CONF_MESSAGE = "message"
+CONF_SUCCESS = "success"
+FEED_ID = "id"
+FEED_NAME = "name"
+FEED_TAG = "tag"
+
+
+def get_options(feeds: list[dict[str, Any]]) -> list[dict[str, Any]]:
+    """Build the selector options with the feed list."""
+    return [
+        {
+            "value": feed[FEED_ID],
+            "label": f"{feed[FEED_ID]}|{feed[FEED_TAG]}|{feed[FEED_NAME]}",
+        }
+        for feed in feeds
+    ]
+
+
+async def get_feed_list(hass: HomeAssistant, url: str, api_key: str) -> dict[str, Any]:
+    """Check connection to emoncms and return feed list if successful."""
+    emoncms_client = EmoncmsClient(
+        url,
+        api_key,
+        session=async_get_clientsession(hass),
+    )
+    return await emoncms_client.async_request("/feed/list.json")
+
+
+class EmoncmsConfigFlow(ConfigFlow, domain=DOMAIN):
+    """emoncms integration UI config flow."""
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> OptionsFlowWithConfigEntry:
+        """Get the options flow for this handler."""
+        return EmoncmsOptionsFlow(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Initiate a flow via the UI."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            self._async_abort_entries_match(
+                {
+                    CONF_API_KEY: user_input[CONF_API_KEY],
+                    CONF_ID: user_input[CONF_ID],
+                    CONF_URL: user_input[CONF_URL],
+                }
+            )
+            result = await get_feed_list(
+                self.hass, user_input[CONF_URL], user_input[CONF_API_KEY]
+            )
+            if not result[CONF_SUCCESS]:
+                errors["base"] = result[CONF_MESSAGE]
+            else:
+                return self.async_create_entry(
+                    title=user_input[CONF_ID], data=user_input
+                )
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.add_suggested_values_to_schema(
+                vol.Schema(
+                    {
+                        vol.Required(CONF_ID): str,
+                        vol.Required(CONF_URL): str,
+                        vol.Required(CONF_API_KEY): str,
+                        vol.Required(CONF_SCAN_INTERVAL, default=30): int,
```

Comment:
scan interval should not be configurable. People can use the `update_entity` service to change the way they poll